packages/@expo/cli/src/prebuild/renameTemplateAppName.ts

```diff
@@ -0,0 +1,178 @@
+ * `getTemplateFilesToRenameAsync()` will check for it automatically).
+ *
+ * The file patterns are formatted as glob expressions to be interpreted by
+ * [fast-glob](https://github.com/mrmlnc/fast-glob). Comments are supported with
+ * the `#` symbol, both in the plain-text file and string array formats.
+ * Whitespace is trimmed and whitespace-only lines are ignored.
+ *
+ * If no rename config has been passed directly to
+ * `getTemplateFilesToRenameAsync()`, and if the template furthermore lacks an
+ * `.expo-rename` file, then this default rename config will be used.
+ *
+ * If you don't want any files in your template to be renamed, simply put an
+ * empty (or comments-only) `.expo-rename` file in the root of your template.
+ */
+export const defaultRenameConfig = [
+  // Common
+  '!**/node_modules',
+  'app.json',
+
+  // Android
+  'android/**/*.gradle',
+  'android/app/BUCK',
+  'android/app/src/**/*.java',
+  'android/app/src/**/*.kt',
+  'android/app/src/**/*.xml',
+
+  // iOS
+  'ios/Podfile',
+  'ios/**/*.xcodeproj/project.pbxproj',
+  'ios/**/*.xcodeproj/xcshareddata/xcschemes/*.xcscheme',
+] as const;
+
+/**
+ * Returns a list of files within a template matched by the resolved rename
+ * config.
+ *
+ * The rename config is resolved in the order of preference:
+ * Config provided as function param > Template-own config > defaultRenameConfig
+ */
+export async function getTemplateFilesToRenameAsync({
+  cwd,
+  /**
+   * An array of patterns following the `.expo-rename` config format. If
+   * omitted, the template-own config will used if present; and if that's
+   * missing too, then we fall back to defaultRenameConfig.
+   * @see defaultRenameConfig
+   */
+  renameConfig: userConfig,
+}: Pick<ExtractProps, 'cwd'> & { renameConfig?: string[] }) {
+  let templateConfig: string[] | null = null;
+
+  if (userConfig) {
+    debug(`Skipping looking for .expo-rename file, as an explicit user config was provided.`);
+  } else {
+    try {
+      const contents = await fs.promises.readFile(path.resolve(cwd, '.expo-rename'), 'utf-8');
+      debug(`Found .expo-rename file: ${contents}`);
+
+      templateConfig = contents.split('\n');
+    } catch (error) {
+      if (!(error instanceof Error && 'code' in error && error.code === 'ENOENT')) {
+        throw new Error(`Unexpected error checking for .expo-rename file`, { cause: error });
+      }
+    }
+  }
+
+  // Resolve the `.expo-rename` config with the following order of preference:
+  // config provided to this function > template-own config > default config
+  let config = userConfig ?? templateConfig ?? defaultRenameConfig;
+
+  // Strip comments, trim whitespace, and remove empty lines.
+  config = config.map((line) => line.split(/(?<!\\)#/, 2)[0].trim()).filter((line) => line !== '');
+
+  return await glob(config, {
+    cwd,
+    // `true` is consistent with .gitignore. Allows `*.xml` to match .xml files
+    // in all subdirs.
+    baseNameMatch: true,
+    dot: true,
+  });
```

Comment:
See my other comment on our usage of fast-glob with user-provided `.expo-rename` templates here.